geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/stats/AccumuloGeoMesaStats.scala

```diff
@@ -72,18 +68,17 @@ class AccumuloGeoMesaStats(val ds: AccumuloDataStore, statsTable: String, val ge
   compactor.run() // schedule initial compaction
 
   override def getCount(sft: SimpleFeatureType, filter: Filter, exact: Boolean): Option[Long] = {
-    lazy val hasDupes = sft.nonPoints && {
-      val indices = AccumuloFeatureIndex.indices(sft, IndexMode.Read)
-      // TODO check for multivalued attribute indices
-      Seq(Z2IndexV1, Z3IndexV2).exists(indices.contains)
+    lazy val query = {
+      // restrict fields coming back so that we push as little data as possible
+      val props = Array(Option(sft.getGeomField).getOrElse(sft.getDescriptor(0).getLocalName))
+      new Query(sft.getTypeName, filter, props)
     }
+    lazy val hasDupes = ds.getQueryPlan(query).exists(_.hasDuplicates)
```

Comment:
they normally don't get evaluated - the `hasDupes` only needs to be checked if `exact` is true, and the `query` only needs to be checked in `hasDupes` is true. I could move query back down into the if statement - not sure why I moved it up tbh...